src/include/transaction/transaction_context.h

```diff
@@ -101,9 +101,10 @@ class TransactionContext {
    * @param slot the TupleSlot inserted
    * @return a persistent pointer to the head of a memory chunk large enough to hold the undo record
    */
-  storage::UndoRecord *UndoRecordForInsert(storage::DataTable *const table, const storage::TupleSlot slot) {
-    byte *const result = undo_buffer_.NewEntry(sizeof(storage::UndoRecord));
-    return storage::UndoRecord::InitializeInsert(result, finish_time_.load(), slot, table);
+  static storage::UndoRecord *UndoRecordForInsert(common::ManagedPointer<TransactionContext> txn,
+                                                  storage::DataTable *const table, const storage::TupleSlot slot) {
+    byte *const result = txn->undo_buffer_.NewEntry(sizeof(storage::UndoRecord));
+    return storage::UndoRecord::InitializeInsert(result, txn->finish_time_.load(), slot, table);
```

Comment:
I changed this to make the arguments needed to call from LLVM IR explicit.  I think it is fine to revert it, but I think whatever we do should be consistent across all the `UndoRecordFor[...]` functions.